c_glib/arrow-glib/input-stream.cpp

```diff
@@ -802,12 +802,14 @@ namespace garrow {
     arrow::Result<int64_t> Read(int64_t n_bytes, void *out) override {
       std::lock_guard<std::mutex> guard(lock_);
       GError *error = NULL;
-      auto n_read_bytes = g_input_stream_read(input_stream_,
-                                              out,
-                                              n_bytes,
-                                              NULL,
-                                              &error);
-      if (n_read_bytes == -1) {
+      gsize n_read_bytes = 0;
+      g_input_stream_read_all(input_stream_,
+                              out,
+                              n_bytes,
+                              &n_read_bytes,
+                              NULL,
+                              &error);
+      if (error) {
```

Comment:
We should use the return value of `g_input_stream_read_all()` instead of `error` to detect error:

```suggestion
      auto success = g_input_stream_read_all(input_stream_,
                                             out,
                                             n_bytes,
                                             &n_read_bytes,
                                             NULL,
                                             &error);
      if (success) {
        return n_read_bytes;
      } else {
        return garrow_error_to_status(error,
                                      arrow::StatusCode::IOError,
                                      "[gio-input-stream][read]");
      }
```

See also: https://docs.gtk.org/gio/method.InputStream.read_all.html